third-party/realdds/doc/notifications.md

```diff
@@ -0,0 +1,86 @@
+
+See also: [device](device.md), [initialization](initialization.md)
+
+
+# Notifications
+
+A topic is used to deliver [flexible](../include/realdds/topics/flexible/) messages from a server to its clients.
+
+> `<topic-root>` /notification
+
+These notifications can fall into many types, encompass almost any data, and happen at ANY time. Several are pre-defined and listed below.
+
+Notifications should always be a JSON (possibly CBOR) object (`{'x': true}` and not just `true`) and identified by an `id` field inside, e.g.:
+
+```JSON
+{
+    "id": "some-message-id",
+    "message": "this is a field value"
+}
+```
+
+Notification content is dependent on the `id`. Fields that are not recognized are ignored by the client. Notifications that are unrecognized are ignored. This allows the server to include additional content as needed.
+
+Messages are broadcast to all clients rather than to a specific one.
+
+
+#### Quality of Service
+
+- Reliability: `RELIABLE`
+- Durability: `VOLATILE`
+
+
+### Logging
+
+As part of the usual runtime of the server, conditions may warrant some kind of output for logging purposes, as is customary in software products.
+
+These may be enabled or disabled via some mechanism, but it is assumed that some basic level of logging will always happen, even if only to communicate error details, state changes, etc.
+
+These logs are output as notifications.
+
+
+#### `log`
+
+Multiple log entries can be packaged together. It is up to the server to decide the frequency of notification. It is recommended that logs be flushed prior to any other notification being sent out, to maintain temporal cohesion in the logs.
+
+- `entries` is an array containing 1 or more log entries
+    - Each entry is a JSON object containing:
+        - `timestamp`: the event occurred (see [timestamps](#timestamps))
+        - `type`: one of `EWID` (Error, Warning, Info, Debug)
```

Comment:
I don't think that we need to shorten this field to a single character, can use the full word, it won't incur a noticeable overhead

Reply:
And yet I feel that the single-letter is as communicative. When logs contains lots (and there will be lots) of entries, it does add up and is repetitive.